torch/csrc/autograd/python_variable.cpp

```diff
@@ -459,6 +473,42 @@ static PyObject * THPVariable_device(THPVariable* self) {
   END_HANDLE_TH_ERRORS
 }
 
+// Returns the type object of the most derived subclass of torch._C._TensorBase that is not an instance of torch.nn.Parameter.
+PyTypeObject *THPVariable_result_ptype(PyObject *self, PyObject *args) {
+    HANDLE_TH_ERRORS
+    PyTypeObject *ptype;
+    PyObject *obj;
+
+    //Initialize on First Call
+    bool initialized = false;
```

Comment:
This should be static!

Reply:
Also, what about thread safety? Should we guard this with an `std::call_once`?

Reply:
This doesn't need to be thread safe because it is in Python code and therefore is GIL protected

Reply:
But yes, it definitely needs to be static lol

Reply:
Oops. I'll fix it.